contracts/src/PDPService.sol

```diff
@@ -253,31 +284,36 @@ contract PDPService {
         return rootId;
     }
 
+    function scheduleRemovals(uint256 setId, uint256[] calldata rootIds) public {
+        require(rootIds.length + scheduledRemovals[setId].length <= MAX_ENQUEUED_REMOVALS, "Too many removals wait for next proving period to schedule");
+        require(proofSetOwner[setId] == msg.sender, "Only the owner can schedule removal of roots");
+        require(proofSetLive(setId), "Proof set not live");
+        uint256 totalDelta = 0;
+        for (uint256 i = 0; i < rootIds.length; i++){
+            totalDelta += rootLeafCounts[setId][rootIds[i]];
+            scheduledRemovals[setId].push(rootIds[i]);
+        }
+        
+        bytes memory extraData = abi.encode(totalDelta, rootIds);
+        _notifyListener(setId, proofSetListener[setId], PDPListener.OperationType.REMOVE_SCHEDULED, extraData);
+    }
+
     // removeRoots removes a batch of roots from a proof set.  Must be called by the proof set owner.
```

Comment:
Move and adapt this comment up to the public scheduleRemovals.

I would also suggest moving the internal remove roots functions down below nextProvingPeriod.